pkgs/development/python-modules/functiontrace/default.nix

```
@@ -15,17 +10,21 @@ buildPythonPackage rec {
     hash = "sha256-3bnxZFq1/D9ntwfv7O2YU6MnKEDWWIG4zX0e3cgCleg=";
   };
 
-  nativeBuildInputs = [
-    setuptools
-    toml
-  ];
+  nativeBuildInputs = [ setuptools toml ];
 
   pythonImportsCheck = [ "functiontrace" ];
 
+  # `functiontrace` needs `functiontrace-server` in its PATH.
+  propagatedBuildInputs = [ functiontrace-server ];
```

I'm not sure how to test that this works on import.  The `postInstall` wrapper certainly does, and that's the much more common way to run `functiontrace` at least.

---

Even with this method, I don't think `propagatedBuildInputs` should be used here.

---

we either need propagatedBuildInputs or makeWrapper, not both. Also please revert the formatting

---

Updated to use only `propagatedBuildInputs`.

---

```suggestion
  postInstall = ''
    wrapProgram $out/bin/functiontrace \
      --set PATH ${lib.makeBinPath [ functiontrace-server ]}
  '';
```
I think this is the more logical way, `functiontrace-server` is a rust package and should not be added to python propagatedBuildInputs.

---

Yes, like ocfox said.